dcos/http.py

```diff
@@ -159,6 +159,10 @@ def _request_with_auth(response,
             else:
                 auth = AUTH_CREDS[creds]
 
+        # if the data is a file buffer, seek to the beginning
+        if 'data' in kwargs and isinstance(kwargs['data'], file):
+            kwargs['data'].seek(0)
+
```

Comment:
I considered putting it in `_request`, but decided against it because `_request_with_auth` is the only place where http retries will happen, and I didn't want to mess with the `kwargs['data']` if it wasn't necessary. Normal `dcos.http.request` calls that don't require retries shouldn't ever have this problem.

Thoughts?